sysml.library/Domain Libraries/Geometry/ShapeItems.sysml

```diff
@@ -0,0 +1,266 @@
+/**
+ * This package provides a model of items that represent basic geometric shapes. 
+ */
+package ShapeItems {
+	private import ScalarValues::Boolean;
+	private import ISQ::*;
+	private import SI::m;
+	private import Occurrences::WithinBoth;
+	private import Objects::*;
+	private import Items::Item;
+	private import SequenceFunctions::isEmpty;
+	
+	/**
+	 * A Line is a Curve that is a straight line of a given length.
+	 */
+	item def Line :> Curve {
+		attribute :>> length;
+		attribute :>> outerSpaceDimension = 1;
+	}
+
+	/**
+	 * A PlanarSurface is a flat Surface with a given area.
+	 */
+	item def PlanarSurface :> Surface {
+		attribute :>> area;
+		attribute :>> outerSpaceDimension = 2;
+	}
+
+	/**
+	 * Shell is the most general Surface shape.
+	 */
+	abstract item def Shell :> Item, StructuredSpaceObject, Surface;
+
+	/**
+	 * Path is the most general Curve shape.
+	 */
+	abstract item def Path :> Item, StructuredSpaceObject, Curve;
+
+	/**
+	 * A Circle is a Path in the shape of a circle of a given radius.
```

Comment:
Wouldn't say "shape", a circle has no boundary so has no Item::shape.  Definition is circular (which actually might be appropriate here!), but I guess avoiding it means saying more than is modeled (eg, points a radius distance from one point).

Reply:
Well, maybe it doesn’t _have_ a shape, but it _is_ a shape — the shape of a disc.